Initial pass at GCM_VSTS_SCOPE (#302)
microsoft/Git-Credential-Manager-Core: docs/configuration.md

---

> ### vstsScope

Reviewer: I would rename this to `azreposPatScopes`

```suggestion
### credential.azreposPatScopes
```

Overrides GCM default scope request when generating a Personal Access Token from Azure DevOps.

> The supported format is one or more [scope values](https://docs.microsoft.com/en-us/vsts/integrate/get-started/authentication/oauth#scopes) separated by whitespace, commas, semi-colons, or pipe `'|'` characters.

Reviewer: New URL, and lets reduce the set of value separators to just commas (pipes and semi-colons might be interpreted by the particular shell flavour the user is using).

```suggestion
The supported format is one or more [scope values](https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth) separated by commas (`,`).
```

> Defaults to `vso.code_write|vso.packaging`; Honored when host is 'dev.azure.com'.

Reviewer: We shouldn't differentiate based on host name.

```suggestion
Default value is `vso.code_write,vso.packaging`.

> 

Reviewer: ```suggestion

**Warning:** `vso.code_write` should always be included in the set of values or else remote Git operations will not be able to complete successfully.

```

```shell

> git config --global credential.microsoft.visualstudio.com.vstsScope vso.code_write|vso.packaging_write|vso.test_write

Reviewer: ```suggestion
git config --global credential.azreposPatScopes vso.code_write,vso.packaging_write,vso.test_write
```

```

> See [GCM_VSTS_SCOPE](Environment.md#gcm_vsts_scope)

Reviewer: ```suggestion
**Also see: [GCM_AZREPOS_PATSCOPES](environment.md#GCM_AZREPOS_PATSCOPES)**
```
